pkg/executor/aggfuncs/builder.go

```diff
@@ -144,6 +144,20 @@ func buildApproxCountDistinct(aggFuncDesc *aggregation.AggFuncDesc, ordinal int)
 	return nil
 }
 
+func getEvalTypeForApproxPercentile(aggFuncDesc *aggregation.AggFuncDesc) types.EvalType {
+	evalType := aggFuncDesc.Args[0].GetType().EvalType()
+	argType := aggFuncDesc.Args[0].GetType().GetType()
+
+	// Sometimes `mysql.EnumSetAsIntFlag` may be set to true, such as when join,
+	// which is unexpected for `buildApproxPercentile` and `mysql.TypeEnum` and `mysql.TypeSet` will return unexpected `ETInt` here,
+	// so here `evalType` are forcibly set to `ETString`.
+	if argType == mysql.TypeEnum || argType == mysql.TypeSet || argType == mysql.TypeBit {
```

Comment:
And the comment for `mysql.TypeBit`?

Reply:
For `mysql.TypeBit`, just same as other aggregate function

Reply:
> And the comment for `mysql.TypeBit`?

ok